llama-index-integrations/vector_stores/llama-index-vector-stores-azureaisearch/llama_index/vector_stores/azureaisearch/base.py

```diff
@@ -638,7 +644,9 @@ def __init__(
             if isinstance(search_or_index_client, SearchIndexClient):
                 # If SearchIndexClient is supplied so must index_name
                 self._index_client = cast(SearchIndexClient, search_or_index_client)
-
+                self._index_client._client._config.user_agent_policy.add_user_agent(
```

Comment:
Thats a lot of private variable access 😅 A little worried about this breaking in the future. But should be ok-ish for now

Reply:
Thanks for the review! Good point about private variable access. 

I propose keeping it as is because:
1. It matches LangChain's implementation pattern
2. It's necessary for tracking LlamaIndex usage in Azure Search telemetry

phoning some friends (@pablocastro @chuwik @pamelafox @marlenezw) any alternative approaches you'd suggest here?

Reply:
Yea totally fine for now. But I would suggest azure updates their python library to support doing this in a more official way 👍🏻

Reply:
> phoning some friends
I agree that depending on private variable access isn't great. You can pass `user_agent` as a named argument to the constructors of `SearchClient` and `SearchIndexClient`, but I don't think there's a way to change it afterwards, we can always ask the SDK team since these pieces are general Azure SDK components.